src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js

```diff
@@ -3,84 +3,174 @@ document.addEventListener('DOMContentLoaded', () => {
   const submitButton = document.querySelector('#settingsFormSubmitButton');
   const cancelButton = document.querySelector('#settingsFormCancelButton');
   const formButtons = Array.from(document.getElementsByClassName('formButton'));
+  const testConnectionButton = document.querySelector('#testConnectionButton');
+  const togglePassword = document.querySelector("#togglePassword");
+  const toggleApi = document.querySelector("#toggleApi");
+  const formBody = document.querySelector('#formBody');
+  const password = document.querySelector("#notificationsPassword");
+  const merchAccount = document.getElementById("merchantAccount");
+  const apiKeyVal = document.getElementById('apiKey');
   const changedSettings = [];
+  testConnectionButton.disabled = true;
 
   function settingChanged(key, value) {
-    const settingIndex = changedSettings.findIndex(
-      (setting) => setting.key === key,
-    );
+    const settingIndex = changedSettings.findIndex(setting => {
+      return setting.key === key;
+    });
 
     if (settingIndex >= 0) {
-      changedSettings[settingIndex] = { key, value };
+      changedSettings[settingIndex] = {
+        key,
+        value
+      };
     } else {
-      changedSettings.push({ key, value });
+      changedSettings.push({
+        key,
+        value
+      });
     }
   }
 
   function enableformButtons() {
-    formButtons.forEach((button) => {
+    for (const button of formButtons) {
```

Comment:
is there a reason this was changed to a for..of loop?
The linter doesn't like that kind of loop because it could have side effects (changes could be made directly on the array elements, rather than creating a copy of the array and manipulating the elements of the copy, which is the case with most array functions like Array.map for example)